### PR TITLE
Add new decorators module

### PR DESCRIPTION
**Pull request by staceyoue**
* New module to access common decorator functions across Toolkit Apps
* Added decorator function to show busy/wait cursor during the execution of a function
* Show busy cursor while filter menu refreshes

### CODE REVIEW

**Review by NorberMV — APPROVED** [review]
Nice feature, LGTM !

File: python/decorators/decorators.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+from functools import wraps
+
+import sgtk
+from sgtk.platform.qt import QtGui, QtCore
+
+
+def wait_cursor(func):
+    """
+    Decorator function that overrides the Qt cursor to show the waiting cursor while a function executes.
```
Comment (julien-lang) [inline python/decorators/decorators.py:19]: Could you split the line at 80 (or 120 I can't remember) chars like the rest of the code, please?

File: python/decorators/decorators.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+#
+# This work is provided "AS IS" and subject to the ShotGrid Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the ShotGrid Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Autodesk, Inc.
+
+from functools import wraps
+
+import sgtk
```
Comment (julien-lang) [inline python/decorators/decorators.py:13]: Is this import required?

File: python/decorators/decorators.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+    :param func: The function to execute
+    :type func: function
+    """
+
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        QtGui.QApplication.setOverrideCursor(QtCore.Qt.WaitCursor)
+        try:
+            response = func(*args, **kwargs)
```
Comment (julien-lang) [inline python/decorators/decorators.py:29]: The intermediary response variable is not used. You could just return here. The finally would run anyway.

**Comment by julien-lang** [comment]
Just to be sure: you don't use this new feature here right? You will use it in a different PR/repo?

**Review by julien-lang — CHANGES_REQUESTED** [review]
I left a few comments to address.

File: python/decorators/decorators.py (hunk 2 above)
Comment (staceyoue) [reply python/decorators/decorators.py]: No it is not, removed.

**Comment by staceyoue** [comment]
> Just to be sure: you don't use this new feature here right? You will use it in a different PR/repo?

Yes, it is used in this PR: https://github.com/shotgunsoftware/tk-framework-qtwidgets/pull/136

**Comment by staceyoue** [comment]
> I left a few comments to address.

Thanks for the review, pushed updates for your comments.

**Review by julien-lang — APPROVED** [review]
LGTM!

**Pull request merged**